src/ImageSharp/MetaData/Profiles/Exif/ExifProfile.cs

```diff
@@ -247,7 +255,33 @@ public byte[] ToByteArray()
             }
 
             var writer = new ExifWriter(this.values, this.Parts);
-            return writer.GetData();
+            return writer.GetData(includeExifIdCode);
+        }
+
+        /// <summary>
+        /// Checks if a byte array start with the Exif Code: ASCII "Exif" followed by two zeros.
+        /// </summary>
+        /// <param name="exifBytes">The byte array to check for the Exif Code.</param>
+        /// <returns>True, if the byte array starts with the Exif Code</returns>
+        private bool StartsWithExifIdCode(byte[] exifBytes)
+        {
+            if (exifBytes.Length < 6)
+            {
+                return false;
+            }
+
+            // The EXIF ID code: ASCII "Exif" followed by two zeros.
+            byte[] exifCode = { 69, 120, 105, 102, 0, 0 };
```

Comment:
Can we avoid this allocation per iteration. Do we know yet whether that identifier is specific to jpeg?

Reply:
i have changed that. 

I am not sure if its jpeg specific or other formats which support Exif also use this. I only can tell for sure that PNG does not use it. Maybe @dlemstra knows more about this?